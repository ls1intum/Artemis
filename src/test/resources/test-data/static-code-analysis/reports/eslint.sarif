{
  "version": "2.1.0",
  "$schema": "http://json.schemastore.org/sarif-2.1.0-rtm.5",
  "runs": [
    {
      "tool": {
        "driver": {
          "name": "ESLint",
          "informationUri": "https://eslint.org",
          "rules": [
            {
              "id": "no-debugger",
              "helpUri": "https://eslint.org/docs/latest/rules/no-debugger",
              "properties": {},
              "shortDescription": {
                "text": "Disallow the use of `debugger`"
              }
            },
            {
              "id": "no-empty",
              "helpUri": "https://eslint.org/docs/latest/rules/no-empty",
              "properties": {},
              "shortDescription": {
                "text": "Disallow empty block statements"
              }
            }
          ],
          "version": "9.20.1"
        }
      },
      "artifacts": [
        {
          "location": {
            "uri": "file:///var/tmp/testing-dir/assignment/src/bubblesort.js"
          }
        },
        {
          "location": {
            "uri": "file:///var/tmp/testing-dir/assignment/src/client.js"
          }
        },
        {
          "location": {
            "uri": "file:///var/tmp/testing-dir/assignment/src/context.js"
          }
        },
        {
          "location": {
            "uri": "file:///var/tmp/testing-dir/assignment/src/mergesort.js"
          }
        },
        {
          "location": {
            "uri": "file:///var/tmp/testing-dir/assignment/src/policy.js"
          }
        }
      ],
      "results": [
        {
          "level": "error",
          "message": {
            "text": "Unexpected 'debugger' statement."
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "file:///var/tmp/testing-dir/assignment/src/client.js",
                  "index": 1
                },
                "region": {
                  "startLine": 21,
                  "startColumn": 9,
                  "endLine": 21,
                  "endColumn": 18
                }
              }
            }
          ],
          "ruleId": "no-debugger",
          "ruleIndex": 0
        },
        {
          "level": "error",
          "message": {
            "text": "Empty block statement."
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "file:///var/tmp/testing-dir/assignment/src/client.js",
                  "index": 1
                },
                "region": {
                  "startLine": 34,
                  "startColumn": 9,
                  "endLine": 34,
                  "endColumn": 11
                }
              }
            }
          ],
          "ruleId": "no-empty",
          "ruleIndex": 1
        }
      ]
    }
  ]
}
