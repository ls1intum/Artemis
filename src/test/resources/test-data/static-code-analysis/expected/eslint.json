{
  "tool": "ESLINT",
  "issues": [
    {
      "filePath": "/var/tmp/testing-dir/assignment/src/client.js",
      "startLine": 21,
      "endLine": 21,
      "startColumn": 9,
      "endColumn": 18,
      "rule": "no-debugger",
      "category": "Lint",
      "message": "Unexpected 'debugger' statement.",
      "priority": "error"
    },
    {
      "filePath": "/var/tmp/testing-dir/assignment/src/client.js",
      "startLine": 34,
      "endLine": 34,
      "startColumn": 9,
      "endColumn": 11,
      "rule": "no-empty",
      "category": "Lint",
      "message": "Empty block statement.",
      "priority": "error"
    }
  ]
}
